packages/react/src/create-config.ts

```diff
@@ -0,0 +1,158 @@
+import {
+  ABSTRACT_API_URL,
+  AccountId,
+  AccountPublicClient,
+  AccountWalletClient,
+  ApiClient,
+  PublicClient,
+  WalletClient,
+  createAccountPublicClient,
+  createAccountWalletClient,
+  createApiClient,
+  createPublicClient,
+  createWalletClient,
+} from '@abstract-money/core'
+import {
+  CosmWasmClient,
+  SigningCosmWasmClient,
+} from '@cosmjs/cosmwasm-stargate'
+import { useMemo } from 'react'
+import { Prettify } from './types/utils'
+
+// Because cosmos-kit does not handle the chain states
+// and requires them to be explicitly passed,
+type CommonProviderArgs = {
+  /** Pass only if you are using cosmos-kit provider, graz doesn't need one. */
+  chainName?: string
+}
+
+export type Provider = {
+  useCosmWasmClient: (arg?: CommonProviderArgs) => CosmWasmClient | undefined
+  useSigningCosmWasmClient: (
+    arg?: CommonProviderArgs,
+  ) => SigningCosmWasmClient | undefined
+  useSenderAddress: (arg?: CommonProviderArgs) => string | undefined
+}
+
+export type CreateConfigParameters = {
+  provider: Provider
+  apiUrl?: string
+}
+
+export function createConfig(parameters: CreateConfigParameters) {
+  const { apiUrl = ABSTRACT_API_URL, provider } = parameters
+
+  /////////////////////////////////////////////////////////////////////////////////////////////////
+  // Set up clients
+  /////////////////////////////////////////////////////////////////////////////////////////////////
+
+  function useApiClient(): ApiClient {
+    return useMemo(
+      () =>
+        createApiClient({
+          apiUrl,
+        }),
+      [apiUrl],
+    )
+  }
+
+  function usePublicClient(
+    ...args: Prettify<Parameters<Provider['useCosmWasmClient']>>
+  ): PublicClient | undefined {
+    const cosmWasmClient = provider.useCosmWasmClient(...args)
+    return useMemo(() => {
+      if (!cosmWasmClient) return undefined
+      return createPublicClient({
+        apiUrl,
+        cosmWasmClient,
+      })
+    }, [apiUrl, cosmWasmClient])
+  }
+
+  function useWalletClient(
+    ...args: Prettify<Parameters<Provider['useSigningCosmWasmClient']>>
+  ): WalletClient | undefined {
+    const signingCosmWasmClient = provider.useSigningCosmWasmClient(...args)
+    const sender = provider.useSenderAddress(...(args as [any]))
```

Comment:
Why is this any array casting necessary?

Reply:
oops, leftover